### PR TITLE
fix: ProblemEditor useEffect at beginning of component; message for block failed to load. TNL-10507

### PR DESCRIPTION
**Pull request by kenclary**
fix: ProblemEditor useEffect at beginning of component; message for block failed to load. TNL-10507

### CODE REVIEW

File: src/editors/containers/ProblemEditor/index.jsx [hunk 1]
```diff
@@ -2,22 +2,26 @@ import React from 'react';
...
   // Redux
   problemType,
   blockFinished,
+  blockFailed,
   studioViewFinished,
   blockValue,
   initializeProblemEditor,
   assetsFinished,
 }) => {
+  React.useEffect(() => initializeProblemEditor(blockValue), []);
```
Comment (jesperhodge) [inline src/editors/containers/ProblemEditor/index.jsx:23]: Now that I am paying attention to the specifics: I think the reason that the useEffect was put after the return statement is to not call the `initializeProblemEditor` method unless block, studioview, and assets finished. The comment says "once data is loaded, init store". It was just the wrong approach. How to call the `initializeProblemEditor` method  conditionally is to add the condition to the callback method inside of the useEffect. However, for the hook to register that the booleans have changed, you also need to put them into the dependency array.

**Review by jesperhodge — CHANGES_REQUESTED** [review]
See my comment; also, there are tests for the new functionality missing which is why the pipeline fails.

File: src/editors/containers/ProblemEditor/index.jsx [hunk 2]
```diff
@@ -29,9 +37,14 @@ export const ProblemEditor = ({
       </div>
     );
   }
-  // once data is loaded, init store
-  React.useEffect(() => initializeProblemEditor(blockValue), []);
-  // TODO: INTL MSG, Add LOAD FAILED ERROR using BLOCKFAILED
+
+  if (blockFailed) {
+    return (
+      <div className="text-center p-6">
```
Comment (jesperhodge) [inline src/editors/containers/ProblemEditor/index.jsx:43]: Should we use a Toast like in the TextEditor, just for consistency?

File: src/editors/containers/ProblemEditor/index.jsx [hunk 3]
```diff
@@ -2,22 +2,30 @@ import React from 'react';
...
   // Redux
   problemType,
   blockFinished,
+  blockFailed,
   studioViewFinished,
   blockValue,
   initializeProblemEditor,
   assetsFinished,
 }) => {
+  React.useEffect(() => {
```
Comment (jesperhodge) [inline src/editors/containers/ProblemEditor/index.jsx:23]: `initializeProblemEditor` seems to be a line that is not covered by tests - probably this is what codecov is complaining about? Up to now we usually have useEffects and such in the hooks file and then tested by mocking out effects and such. That could be a solution here if you don't want to use new tooling.

I'm just trying to suggest anything that could fix the code coverage problem, but go ahead and fix that in any way you see fit, I don't have any preference.
- Wile the place where this useEffect logic is right now is not consistent with the rest of the application, that is not due to you and not something we need to fix now.

Comment (kenclary) [reply src/editors/containers/ProblemEditor/index.jsx:23]: isn't "test `initializeProblemEditor`" the whole point of the (currently excluded) first bullet in the ticket?

**Review by jesperhodge — APPROVED** [review]
The comments I left earlier are just for the pipeline and such, so this is approved from my side, just needs the pipeline figured out and so on.

**Pull request merged**